pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -155,6 +155,39 @@ func NewStatus(code Code, reasons ...string) *Status {
 	}
 }
 
+type PluginToStatus map[string]*Status
+
+func (p PluginToStatus) Merge() *Status {
+	if len(p) == 0 {
+		return nil
+	}
+
+	finalStatus := NewStatus(Success)
```

Comment:
what about defining a "precedence" array or map. Then simply take the status with higher precedence. It seems easier to read too.

Reply:
I am not sure it is necessary, the order is clear in the last block.